core/src/main/scala/org/http4s/otel4s/middleware/ServerMiddleware.scala

```diff
@@ -91,37 +99,59 @@ object ServerMiddleware {
         allowedResponseHeaders,
         routeClassifier,
         serverSpanName,
-        additionalRequestTags,
-        additionalResponseTags,
+        additionalRequestAttributes,
+        additionalResponseAttributes,
         urlRedactor,
         doNotTrace,
       )
 
+    /** Sets which request headers are allowed to made into `Attribute`s. */
     def withAllowedRequestHeaders(allowedHeaders: Set[CIString]): ServerMiddlewareBuilder[F] =
       copy(allowedRequestHeaders = allowedHeaders)
+
+    /** Sets which response headers are allowed to made into `Attribute`s. */
     def withAllowedResponseHeaders(allowedHeaders: Set[CIString]): ServerMiddlewareBuilder[F] =
       copy(allowedResponseHeaders = allowedHeaders)
+
+    /** Sets how to determine the route within the application from a request.
+      * A value of `None` returned by the given function indicates that the
+      * route could not be determined.
+      */
     def withRouteClassifier(
         routeClassifier: Request[F] => Option[String]
     ): ServerMiddlewareBuilder[F] =
       copy(routeClassifier = routeClassifier)
+
+    /** Sets how to derive the name of a server span from a request. */
     def withServerSpanName(serverSpanName: Request[F] => String): ServerMiddlewareBuilder[F] =
       copy(serverSpanName = serverSpanName)
-    def withAdditionalRequestTags(
-        additionalRequestTags: Request[F] => Seq[Attribute[_]]
+
+    /** Sets how to derive additional `Attribute`s from a request to add to the
+      *  server span.
+      */
+    def withAdditionalRequestAttributes(
+        additionalRequestAttributes: Request[F] => Seq[Attribute[_]]
     ): ServerMiddlewareBuilder[F] =
-      copy(additionalRequestTags = additionalRequestTags)
-    def withAdditionalResponseTags(
-        additionalResponseTags: Response[F] => Seq[Attribute[_]]
+      copy(additionalRequestAttributes = additionalRequestAttributes)
+
+    /** Sets how to derive additional `Attribute`s from a response to add to the
+      *  server span.
+      */
+    def withAdditionalResponseAttributes(
+        additionalResponseAttributes: Response[F] => Seq[Attribute[_]]
     ): ServerMiddlewareBuilder[F] =
-      copy(additionalResponseTags = additionalResponseTags)
+      copy(additionalResponseAttributes = additionalResponseAttributes)
+
+    /** Sets how to redact URLs before turning them into `Attribute`s. */
     def withUrlRedactor(urlRedactor: UriRedactor): ServerMiddlewareBuilder[F] =
       copy(urlRedactor = urlRedactor)
+
+    /** Sets how to determine when not to trace a request and its response. */
```

Comment:
Not a new problem, but this API seems like a case of boolean blindness.

Reply:
yeah, I'm not a fan either. might try to brainstorm a refactor

Reply:
it'll be a separate commit/PR anyway though